warehouse/query-core/src/main/java/datawave/query/planner/DefaultQueryPlanner.java

```diff
@@ -3229,8 +3259,254 @@ public static Date getEndDateForIndexLookup(Date endDate) {
 
     @Override
     public void finalize() {
-        if (null != builderThread) {
-            builderThread.shutdown();
+        if (null != executor) {
+            executor.shutdown();
         }
     }
+
+    protected CompositeMetadata getCompositeMetadata() {
+        if (compositeMetadata == null && compositeMetadataCallable != null) {
+            TraceStopwatch stopwatch = stageStopWatch.newStartedStopwatch(compositeMetadataCallable.stageName());
+            try {
+                while (compositeMetadata == null) {
+                    compositeMetadata = compositeMetadataFuture.get(concurrentTimeoutMillis, TimeUnit.MILLISECONDS);
+                }
+            } catch (InterruptedException | ExecutionException | TimeoutException e) {
+                log.error("Failed to fetch CompositeMetadata");
+                log.error(e.getStackTrace());
```

Comment:
these lines can be combined so that log parsing can process this as a single event (all other cases as well)

Reply:
combined in next commit